### PR TITLE
VN-517 Improve idle mining logic

### PR DESCRIPTION
**Pull request by angels2it**
## Ticket references
- [VN-517](https://freedom.myjetbrains.com/youtrack/issue/VN-517)

## Summary of changes

* Add logic to skip idle mining if CPU usage greater than 50%
* Add logic to skip udle mining if has an application in exclusion list is running
* A page to help user config application exclusion list
* Add language for VN-517

## Checklist

- [ ] bugfix
- [x] new feature
- [ ] breaking change
- [ ] added unit tests
- [ ] changes introduced has been discussed and approved
- [ ] requires manual testing
- [ ] documentation has been updated

## Testing

* Enable idle mining

Case 1:
- Select application in config -> to said idle mining does not run while it's running
- Starting selected app
- let computer idle 30s
Expected: miner does not start

Case 2:
- Select application in config -> to said idle mining does not run while it's running
- Stop selected app
- let computer idle 30s
Expected: idle logic trigger start mining action

For CPU case, You may need to update config of CPU_MAX_LOAD to small value for testing or try to burn you CPU usage to > 50 percent :D

### CODE REVIEW

File: cs/index.json [hunk 1]
```diff
@@ -77,7 +77,13 @@
     "gpu_mining_description" : "Automatically start GPU mining when start mining",
     "mining_options_changed": "Please restart mining (click Stop, Start) to apply the changes",
     "mine_even_projected_profit_lost": "Mine coin even when estimated earning is projected a profit lost?",
-    "mine_even_projected_profit_lost_description": "Yes to mine all coins in your preferred list, No to mine filtered profitable cryptoes only"
+    "mine_even_projected_profit_lost_description": "Yes to mine all coins in your preferred list, No to mine filtered profitable cryptoes only",
+    "exclude_applications": {
+      "title": "Applications exclusion",
+      "description": "Select application(s) that you want to skip idle mining while it's running",
+      "filter": "Search an application",
+      "reload_hint": "If the application does not has in below list, Try to start it & click refresh icon"
```
Comment (pandafox) [inline cs/index.json]: If the application you want to add is not listed, try to start it first and come back and refresh the list.

File: cs/index.json [hunk 2]
```diff
@@ -85,6 +85,7 @@
       "reload_hint": "If the application does not has in below list, Try to start it & click refresh icon",
       "select_application_button": "Do not mining if my applications are running",
       "add_hint": "Add an application by select it from file dialog",
+      "clear_hint": "Clear all selected applications",
```
Comment (pandafox) [inline cs/index.json]: Would this be clear, or remove all selected applications?

File: zh/index.json [hunk 3]
```diff
@@ -77,7 +77,18 @@
...
+    "exclude_applications": {
+      "title": "Applications exclusion",
+      "description": "Select application(s) that you want to skip idle mining while it's running",
+      "filter": "Search an application",
+      "reload_hint": "If the application you want to add is not listed, try to start it first and come back and refresh the list.",
+      "select_application_button": "Do not mining if my applications are running",
+      "add_hint": "Add an application by select it from file dialog",
+      "remove_hint": "Remove all selected applications",
+      "selected_applications": "Selected applications",
+      "running_applications": "Running applications"
```
Comment (pandafox) [inline zh/index.json]:
```
 "title": "排除應用程序",
      "description": "當下列應用程序運行時，空閒挖掘將不會啟動",
      "filter": "“搜索應用程序",
      "reload_hint": "如果未列出要添加的應用程序，請先嘗試啟動它，然後返回並刷新列表。",
      "select_application_button": "“如果我的應用程序正在運行，不要挖掘",
      "add_hint": "通過從文件對話框中選擇它來添加應用程序",
      "remove_hint": "刪除所有選定的應用程序",
      "selected_applications": "選定的應用程序",
      "running_applications": "正在運行的應用程序"
```

File: zh/index.json [hunk 4]
```diff
@@ -130,8 +141,12 @@
       "add_to_exclusion_button": "自動添加採礦工具文件夾到排除列表",
       "add_miners_folder_to_exclusion_list_successful": "採礦工具文件夾已成功添加到排除列表中",
       "add_miners_folder_to_exclusion_list_error": "將採礦工具文件夾添加到排除列表時失敗，請再試一次或手動執行",
+      "auto_add_miners_folder_to_exclusion_list_error_title": "Miner exclusion error",
+      "auto_add_miners_folder_to_exclusion_list_error_message": "Error while adding miners folder to Windows Defender's exclussion list, Please click button below to try again or try to restart Crypto Miner",
       "windows_defender": "Windows Defender",
-      "other_virus_software": "其他的防毒軟體"
+      "other_virus_software": "其他的防毒軟體",
+      "restart_crypto_miner_button": "Restart Crypto Miner",
+      "exclusion_confirm_button": "I have added miners folder to my Antivirus software"
```
Comment (pandafox) [inline zh/index.json]: shouldn't we replace crypto miner, windows defender with the translation key directly so we can use the same wording?

Additionally, 
```
"exclusion_confirm_button": "I have added miners folder to my Antivirus software's exclusion list."
```

```
      "auto_add_miners_folder_to_exclusion_list_error_title": "採礦工具錯誤排除",
      "auto_add_miners_folder_to_exclusion_list_error_message": "採礦工具文件夾添加到Windows Defender的排除列表時出錯，請單擊下面的按鈕再試一次或嘗​​試重新啟動Crypto Miner",
      "restart_crypto_miner_button": "重啟Crypto Miner",
      "exclusion_confirm_button": "我已將採礦工具文件夾添加到我的防病毒軟件排除列表"
```

**Review by pandafox — CHANGES_REQUESTED** [review]

File: zh/index.json [hunk 5]
```diff
@@ -123,15 +134,20 @@
     "error_hash_verify_failed": "下載挖掘工具軟體時出現錯誤，請重試",
     "error_unzip": "解壓縮zip文件時出錯，請重試",
     "initializing": "檢查採礦工具資訊",
+    "checking_windows_defender_config": "Checking Windows Defender configuration",
```
Comment (angels2it) [inline zh/index.json]: @pandafox  Please help translate this key

Comment (pandafox) [reply zh/index.json]:
```
"checking_windows_defender_config": "正在檢查Windows Defender設定",
```

**Review by pandafox — APPROVED** [review]

**Pull request merged**